src/lib.rs

```diff
@@ -2236,4 +2371,216 @@ mod tests {
+        #[case] max: i64,
+        #[case] error: Option<&str>,
+    ) {
+        let config = ConfigBuilder::default()
+            .user(user)
+            .build()
+            .expect("Failed to build configuration");
+        let image_config = ImageConfigurationBuilder::default()
+            .config(config)
+            .build()
+            .expect("Failed to build image configuration");
+        let validation_request = ValidationRequest {
+            settings: Settings {
+                run_as_user: RuleStrategy {
+                    rule,
+                    ranges: vec![IDRange { min, max }],
+                    ..Default::default()
+                },
+                ..Default::default()
+            },
+            request: KubernetesAdmissionRequest {
+                ..Default::default()
+            },
+        };
+        let security_context = Some(SecurityContext {
+            run_as_user: Some(min),
+            ..Default::default()
+        });
+        let result =
+            enforce_run_as_user_rule(security_context, &validation_request, Some(image_config));
+        match error {
+            Some(e) => {
+                let error = result.expect_err("Expected error but got success");
+                assert_eq!(error.to_string(), e,);
+            }
+            None => {
+                assert!(
+                    result.is_ok(),
+                    "Expected success but got error: {}",
+                    result.expect_err("Cannot get error").to_string()
+                );
+            }
+        }
+    }
+
+    #[rstest]
+    #[case("65533", Rule::MustRunAs, 65533, 65533, None)]
+    #[case("65534", Rule::MustRunAs, 65533, 65533, None)]
+    #[case("65533:65533", Rule::MustRunAs, 65533, 65533, None)]
+    #[case(
+        "myuser:65533",
+        Rule::MustRunAs,
+        65533,
+        65533,
+        Some("Invalid user ID in the container image configuration: \"myuser\"")
+    )]
+    #[case(
+        "myuser:mygroup",
+        Rule::MustRunAs,
+        65534,
+        65534,
+        Some("Invalid user ID in the container image configuration: \"myuser\"")
+    )]
+    #[case(
+        "65533:mygroup",
+        Rule::MustRunAs,
+        65533,
+        65533,
+        Some("Invalid group ID in the container image configuration: \"mygroup\"")
+    )]
+    #[case(
+        "65533:65534",
+        Rule::MustRunAs,
+        65533,
+        65533,
+        Some("Group ID defined in the container image is outside defined ranges")
+    )]
+    #[case(
+        "65533:65532",
+        Rule::MustRunAs,
+        65533,
+        65533,
+        Some("Group ID defined in the container image is outside defined ranges")
+    )]
+    fn enforce_run_as_group_rule_in_container_image(
+        #[case] user: &str,
+        #[case] rule: Rule,
+        #[case] min: i64,
+        #[case] max: i64,
+        #[case] error: Option<&str>,
+    ) {
+        let config = ConfigBuilder::default()
+            .user(user)
```

Comment:
shouldn't this be group?

Reply:
Ah, it just the variable name. In the container image configuration there is just one field called `user` to store the user and group like this : `user:group`. But I'll update the variable name to make that explicit. ;)

Reply:
Thanks, resolving!